tests/fpsemi-examples.cpp

```diff
@@ -99,6 +99,115 @@ namespace libsemigroups {
         }
       }
     }
+    void add_Iwahori_full_transformation_monoid_relations(
+        std::vector<relation_type>& result,
+        size_t                      n,
+        size_t                      m) {
+      // This function adds the full transformation monoid relations due to
+      // Iwahori, from Section 9.3, p161-162, (Ganyushkin + Mazorchuk),
+      // expressed in terms of the generating set {pi_2, ..., pi_n,
+      // epsilon_{12}} using the notation of that chapter.
+      // https://link.springer.com/book/10.1007/978-1-84800-281-4
+
+      // The argument m specifies the letter value the idempotent e12
+      // corresponds to. When adding these relations for the full transformation
+      // monoid presentation, we want m = n - 1. For the partial transformation
+      // monoid presentaiton, we want m = n.
+
+      // It is useful to have this as a separate function, to avoid computing
+      // duplicate presentations for the underlying symmetric group, when
+      // combining relations from the symmetric inverse and full transformation
+      // monoids.
+
+      word_type              e12 = {m};
+      std::vector<word_type> pi;
+      for (size_t i = 0; i <= n - 2; ++i) {
+        pi.push_back({i});
+      }
+
+      // The following expresses the epsilon idempotents in terms of the
+      // generating set
+      auto eps = [&e12, &pi](size_t i, size_t j) -> word_type {
+        LIBSEMIGROUPS_ASSERT(i != j);
+        if (i == 1 and j == 2) {
+          return e12;
+        } else if (i == 2 and j == 1) {
+          return pi[0] * e12 * pi[0];
+        } else if (i == 1) {
+          return pi[0] * pi[j - 2] * pi[0] * e12 * pi[0] * pi[j - 2] * pi[0];
+        } else if (j == 2) {
+          return pi[i - 2] * e12 * pi[i - 2];
+        } else if (j == 1) {
+          return pi[0] * pi[i - 2] * e12 * pi[i - 2] * pi[0];
+        } else if (i == 2) {
+          return pi[j - 2] * pi[0] * e12 * pi[0] * pi[j - 2];
+        }
+        return pi[i - 2] * pi[0] * pi[j - 2] * pi[0] * e12 * pi[0] * pi[j - 2]
+               * pi[0] * pi[i - 2];
+      };
+
+      auto transp = [&pi](size_t i, size_t j) -> word_type {
+        LIBSEMIGROUPS_ASSERT(i != j);
+        if (i > j) {  // FIXME Want to do: return transp(j, i);, but can't do
```

Comment:
*Todo:* I'd prefer to achieve the same here with `return transp(j, i)`, but this won't compile. I understand this is something to do with the lambda function not being within its own scope, and I'll figure out what the best way to sort this is.

Reply:
I understand that I can fix this by adding `transp` as a third argument to itself, per [this](https://artificial-mind.net/blog/2020/09/12/recursive-lambdas) solution. However, I think this is way less readable than the temporary 'manual' swap I put in, so I won't. I could also get away with not setting `i < j`, but I think doing this makes the relations easier to read.

I'll leave it as is for now.